src/Features/Core/IntroduceVariable/AbstractIntroduceVariableService.State.cs

```diff
@@ -87,7 +87,10 @@ public State(TService service, SemanticDocument document)
                     return false;
                 }
 
-                this.IsConstant = this.Document.SemanticModel.GetConstantValue(this.Expression, cancellationToken).HasValue;
+                // Interpolated strings are not constants themselves, even if they were composed entirely from constants.
+                this.IsConstant = _service.IsInterpolatedStringExpression(this.Expression)
+                    ? false
+                    : this.Document.SemanticModel.GetConstantValue(this.Expression, cancellationToken).HasValue;
```

Comment:
@gafter: I don't think SemanticModel.GetConstantValue()` should ever return a value for an `InterpolatedStringExpression`.  Is the right fix here to fix the API and just add these tests?